vignettes/secuTrialR-package-vignette.Rmd

```diff
@@ -1,172 +1,570 @@
+# load silently
+suppressMessages(library(secuTrialR))
+# show secuTrialR version
+installed.packages()["secuTrialR", "Version"]
+```
+
+# The CTU05 dataset
+
+Before we continue with the functionalities let's briefly talk about the test data which is
+delivered as a part of the package. We refer to it as the `CTU05` (clinical trial unit project 05)
+data. This dataset has been fabricated for demonstration purposes only and is not
+real clinical data. Principally it is made up of eight forms. These are called "surgeries",
+"baseline", "outcome", "treatment", "allmedi", "studyterminat", "ae" and "sae".
+You will see these names again later when the data has been read into R.
+The project setup includes most data types implementable in secuTrial. It is, however,
+not exhaustive. Since the data is delivered with the installation of the `secuTrialR`
+package we can point to it via the `system.file()` function.
+\vspace{5pt}
+
+```{r}
+ctu05_data_location <- system.file("extdata", "sT_exports", "export_options",
+                                   "s_export_CSV-xls_CTU05_20191003-144349_all_info.zip",
+                                   package = "secuTrialR")
+```
+
+If you work on your own datasets you can specify a path as a regular character string without
+using `system.file()`.
+
+\newpage
+
+# secuTrial export options
+
+Prior to reading your data into R you need to export it with the secuTrial ExportSearchTool.
+We suggest exporting **non-rectangular**, **zipped**, **English** data with 
+**reference values** stored **in a separate table** including 
+**Add-IDs**, **centre information**, **form status**, **project setup**
+and **without duplicated meta data**. Furthermore, it is important to use
+**"CSV format"**/**"CSV format for MS Excel"** and suggested to select **UTF-8** encoding.
+Most of these options are truly optional and reading your data should
+work even with differences from the above specifications.
+
+# Reading a secuTrial data export into R
+
+There is one principle function to read your data (i.e. `read_secuTrial()`).
+Below you can see it in action with the `CTU05` dataset.
+\vspace{5pt}
+
+```{r}
+ctu05_data <- read_secuTrial(data_dir = ctu05_data_location)
+```
+
+If the "Read export successfully." message appears your data was correctly read.
+
+# The `secuTrialdata` object
+
+If you inspect the `class()` of `ctu05_data` you will find that it is a `secuTrialdata` object.
+\vspace{5pt}
+
+```{r}
+class(ctu05_data)
+```
+
+Really this is only a `list` containing all the information from your secuTrial export.
+\vspace{5pt}
+
+```{r}
+typeof(ctu05_data)
+```
+
+## The data tables in the `secuTrialdata` object
+
+We have implemented a custom variation of the `print()` function for `secuTrialdata` objects.
+
+```{r}
+print(ctu05_data)
+```
+
+It shows you where the export archive of your `secuTrialdata` object is located, tells you which
+tables (i.e. `table`) it contains, what the source files (i.e. `original_name`) are and
```

Comment:
Later on in the vignette, we refer to these tables as "data tables". Maybe call it "data tables" here as well?